src/superblock.rs

```diff
@@ -104,6 +112,19 @@ impl Superblock {
                 32
             };
 
+        let journal_inode = if compatible_features
+            .contains(CompatibleFeatures::HAS_JOURNAL)
+        {
+            // This cannot fail because of the call to
+            // `check_incompat_features` above.
```

Comment:
nit: I'm struggling to figure out why this assert is here. Protecting against some future change, I think, but I can't quite see what it is. Could the comment explain a little more?

I think this is so that if we take the check for `SEPARATE_JOURNAL_DEVICE`/`INCOMPAT_JOURNAL_DEV` out of `check_incompat_features` there'll be a failure if we try to `HAS_JOURNAL & SEPARATE_JOURNAL_DEVICE`? But in that case I feel like I'd expect a `Corrupt` error, not an assert. Am I misunderstanding?

Reply:
Updated comment to hopefully be a little clearer.

> I think this is so that if we take the check for SEPARATE_JOURNAL_DEVICE/INCOMPAT_JOURNAL_DEV out of check_incompat_features there'll be a failure if we try to HAS_JOURNAL & SEPARATE_JOURNAL_DEVICE? But in that case I feel like I'd expect a Corrupt error, not an assert. Am I misunderstanding?

I am not 100% sure, but I think if a separate journal device is in use then both the `HAS_JOURNAL` and  `SEPARATE_JOURNAL_DEVICE` bits will be set, so that wouldn't indicate corruption. The purpose of the assert is to help out Future Developer who wants to add support for a separate journal device; this assert clearly shows a place where the current code cannot handle a separate journal device.